src/main/java/lotto/Controller/Controller.java

```diff
@@ -0,0 +1,104 @@
+package lotto.Controller;
+
+import java.util.ArrayList;
+import lotto.Model.GameModel;
+import lotto.Model.ScoreBoard;
+import lotto.Model.LottoData;
+import lotto.View.Input;
+import lotto.View.Output;
+
+public class Controller {
+    private GameModel gameModel;
+
+    public void playingGame(){
+        startGame();
+        proceedGame();
+        finshGame();
+    }
+
+    public Controller(){
+        gameModel = new GameModel();
+    }
+
+    private void startGame() {
+        insertPlayerMoney();
+        issuanceLotto();
+    }
+    private void proceedGame(){
+        createAnswerLotto();
+        createBonusNumber();
+    }
+
+    private void issuanceLotto(){
+        gameModel.lottoIssuance();
+        ArrayList<LottoData> playerLottos = gameModel.transferPlayerLottoDatas();
+        Output.printLottos(playerLottos);
+    }
+
+    private void insertPlayerMoney(){
+        boolean errorChecker = true;
+
+        while (errorChecker) {
+            try {
+
+                Output.printEnterPurchaseMount();
+                Integer playerInsert = Input.insertInteger();
+                gameModel.initGameModel(playerInsert);
+
+                errorChecker=false;
+            }catch (IllegalArgumentException error){
+                Output.printError(error.getMessage());
+            }
+        }
+
+    }
```

Comment:
```suggestion
    private void insertPlayerMoney(){
            try {
                Output.printEnterPurchaseMount();
                Integer playerInsert = Input.insertInteger();
                gameModel.initGameModel(playerInsert);

            }catch (IllegalArgumentException error){
                Output.printError(error.getMessage());
                insertPlayerMoney();
            }
        }

    }
```
재귀를 사용하면 반복문을 사용하지 않아 indent를 줄일 수 있습니다! 😁
물론 성능은 반복문이 좀 더 좋을 것 같지만 저는 가독성을 생각해서 재귀를 사용하는 편 입니다! ㅎㅎ

Reply:
> 재귀를 사용하면 반복문을 사용하지 않아 indent를 줄일 수 있습니다! 😁 물론 성능은 반복문이 좀 더 좋을 것 같지만 저는 가독성을 생각해서 재귀를 사용하는 편 입니다! ㅎㅎ

학부때 교수님이 재귀함수가 안좋은거라고  강조하신 경험이 있어서 재귀는 피하는 경향이 있는것같아요 ㅋㅋ
가독성 측면에서는 확실히 좋네요 !!